### ISSUE TITLE
Vagrant provision Error: curl -s http://getcomposer.org/installer

### ISSUE THREAD

**Issue by mans202** [puphpet/puphpet#1374]
==> default: Error: curl -s http://getcomposer.org/installer | php returned 255 instead of one of [0
]
==> default: Error: /Stage[main]/Composer/Exec[download_composer]/returns: change from notrun to 0 f
ailed: curl -s http://getcomposer.org/installer | php returned 255 instead of one of [0]
==> default: Notice: /Stage[main]/Composer/File[/usr/local/bin/composer]: Dependency Exec[download_c
omposer] has failures: true
==> default: Warning: /Stage[main]/Composer/File[/usr/local/bin/composer]: Skipping because of faile
d dependencies
==> default: Notice: /Stage[main]/Main/Exec[dotfiles]/returns: executed successfully
==> default: Notice: /Stage[main]/Pyenv/Vcsrepo[/usr/local/pyenv]/revision: revision changed '3cde54
0ca00208fbfd0a5e7708c3053bc8cca9f5' to 'v0.4.0-20140404'
==> default: Notice: /Stage[main]/Main/Vcsrepo[/usr/share/drush]/revision: revision changed '6ad8c93
c79470c91b7589969eba5d3de08526e61' to '7.0.0-alpha8'
==> default: Notice: /Stage[main]/Main/Composer::Exec[drush]/Exec[composer_update_drush]: Dependency
 Exec[download_composer] has failures: true
==> default: Warning: /Stage[main]/Main/Composer::Exec[drush]/Exec[composer_update_drush]: Skipping
because of failed dependencies
==> default: Notice: /Stage[main]/Main/Exec[first drush run]: Dependency Exec[download_composer] has
 failures: true
==> default: Warning: /Stage[main]/Main/Exec[first drush run]: Skipping because of failed dependenci
es
==> default: Notice: /Stage[main]/Main/File[symlink drush]: Dependency Exec[download_composer] has f
ailures: true
==> default: Warning: /Stage[main]/Main/File[symlink drush]: Skipping because of failed dependencies

==> default: Notice: /Stage[main]/Main/Php_pecl_mod[pecl_http]/Puphpet::Php::Pecl[pecl_http]/Php::Pe
cl::Module[pecl_http]/Exec[pecl-pecl_http]/returns: executed successfully
==> default: Notice: Finished catalog run in 47.72 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

**Comment by wilfredwaltman** [comment]
+1, I get exactly the same error.

**Comment by jtreminio** [comment]
@wilfredwaltman this was fixed, need to regenerate your config, or apply this patch:

https://github.com/puphpet/puppet-composer/commit/c4f9478869296df9344c8d3940ee16666dab6b18

**Comment by wilfredwaltman** [comment]
@jtreminio, thanks for that quick reply.
Then I probably have another issue:

==> default: Error: curl -sS https://getcomposer.org/installer | php returned 1 instead of one of [0]
==> default: Error: /Stage[main]/Composer/Exec[download_composer]/returns: change from notrun to 0 failed: curl -sS https://getcomposer.org/installer | php returned 1 instead of one of [0]
==> default: Warning: /Stage[main]/Composer/File[/usr/local/bin/composer]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Puphpet::Php::Drush/Composer::Exec[drush]/Exec[composer_install_drush]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Puphpet::Php::Drush/Exec[first drush run]: Skipping because of failed dependencies
==> default: Warning: /Stage[main]/Puphpet::Php::Drush/File[symlink drush]: Skipping because of failed dependencies

I will continue my search...

**Comment by jtreminio** [comment]
@wilfredwaltman ssh into the VM and run that command manually, see what error it gives you

**Comment by wilfredwaltman** [comment]
No error then, that's what surprises me. But then if I do a vagrant-reload --provision again it is still there.

**Comment by resurtm** [comment]
Any fix/workaround for this?

**Comment by Sharkfinger** [comment]
I just got this error this morning. Changing:

`curl -s http://getcomposer.org/installer | ${composer::php_bin}`

To 
`curl -sS https://getcomposer.org/installer | ${composer::php_bin}`

Did not work for me either

**Comment by shanept** [comment]
Sharkfinger is on to it. I've figured this one out for myself unfortunately, just note that there are different settings for Debian as opposed to RedHat/CentOS. I am using CentOS and changed the setting for Debian, got me confused for a bit.

**Comment by red-smeg** [comment]
has anyone found a solution that will work without doing it manually

**Comment by shanept** [comment]
@red-smeg, see my pull request at puphpet/puppet-composer#2

**Comment by MisterPeepers** [comment]
Just a note, none of the above works, not even @shanept's pull request above :(

**Comment by pdbreen** [comment]
@MisterPeepers try uploading your config and regen everything with the latest puphpet.  That resolved it for good for me.

**Comment by tlartaud** [comment]
@MisterPeepers You are commenting a very old and fixed issue. Not sure our problem is related to this. Do you the exactly same output error message ? Personnally, i just ran into an issue about composer installation failure, I googled the error output and fall on this page. My issue was coming from a config.yaml update from puphpet. When you update from an old version to this new one, there's a new feature that you have to configure before generating the archive.
In Deploy Targets -> locally -> machine -> add a machine and also make sure you add some pair ports for the 22 port or this port will be blocked by IP Table, and composer need it for installing.
